packages/core/core/src/core-api/routes/validation/utils.ts

```diff
@@ -155,10 +190,12 @@ export const safeGlobalRegistrySet = (id: string, schema: z.ZodType) => {
 export const safeSchemaCreation = (id: string, callback: () => z.ZodType) => {
   const { _idmap: idMap } = z.globalRegistry;
 
-  const existsInGlobalRegistry = idMap.has(id);
+  const transformedId = transformUidToValidOpenApiName(id);
+
+  const existsInGlobalRegistry = idMap.has(transformedId);
 
   if (existsInGlobalRegistry) {
-    return idMap.get(id) as z.ZodType;
+    return idMap.get(transformedId) as z.ZodType;
```

Comment:
`.has` doesn't work as a typeguard, we can probably replace it and do this instead to avoid `as`

```suggestion
    const item = idMap.get(transformedId);
    if (!item) {
      throw new Error(`Schema with id ${transformedId} not found in global registry`);
    }
    return item;
```

Maybe log a warning instead of throwing though? That might be better so we don't block the entire file generation from a small error.

Reply:
Done in [`4c01fe6` (#23833)](https://github.com/strapi/strapi/pull/23833/commits/4c01fe6b1546b49c95dad41483de1d3d9a8d296a) ??